src/tree/identity_tree.rs

```diff
@@ -20,10 +20,15 @@ pub type Leaves = HashMap<LeafIndex, Hash>;
 pub type StorageUpdates = HashMap<NodeIndex, Hash>;
 
 pub struct IdentityTree<S> {
+    /// Densely allocated (and in some cases cached on disk) merkle tree
     pub tree: CascadingMerkleTree<PoseidonHash, S>,
-    pub tree_updates: BTreeMap<Root, StorageUpdates>,
-    // Hashmap of root hash to nonce
-    pub roots: HashMap<Hash, usize>,
+
+    /// Temporary storage of tree updates for a given root
+    /// the updates are stored here until they are applied to all observed chains
+    /// at which point they will be removed from this map and applied to the canonical tree
+    pub tree_updates: Vec<(Hash, StorageUpdates)>,
```

Comment:
Most important change in state representation is this. Previously we mapped a root hash to a nonce. Now the ordering here is implicit and maintained by the logic in `append_updates` - each entry in this vec must either match the previous update via pre/post roots or the pre root must match `tree.root()`